src/clr-angular/data/tree-view/models/tree-node-model.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ClrSelectedState } from './selected-state';
+import { BehaviorSubject } from 'rxjs';
+
+export abstract class TreeNodeModel<T> {
+  selected = new BehaviorSubject<ClrSelectedState>(ClrSelectedState.UNSELECTED);
+  model: T | null;
+  // For some reason the polymorphic this doesn't work for either of my use cases. Dammit, Typescript.
+  // So I'm resorting to forcing override with more precise types by marking these abstract.
+  abstract parent: TreeNodeModel<T> | null;
+  abstract children: TreeNodeModel<T>[];
+
+  loading = false;
+
+  destroy() {
```

Comment:
Why wouldn't you want to recursively complete the children observables here? Otherwise you have to call each individually.

Also this is only tested in the other extended classes.

Reply:
Yes, I'm missing a unit test here. As I said on Slack, I was just trying to have minimal test coverage to meet the deadline, but I can add one for this.

I initially had it recursively detroy, but not only will the components do it for you (which isn't really an argument for this PR), even in the pure model it makes sense to want to destroy a parent without destroying the children. So I decided to keep it local. I don't feel strongly either way.